src/main/java/dev/tycho/stonks/managers/DatabaseHelper.java

```diff
@@ -857,6 +858,246 @@ public void changeVerification(Player player, String companyName, boolean newVer
+            } catch (SQLException e) {
+              e.printStackTrace();
+              player.sendMessage(ChatColor.RED + "SQL EXCEPTION TELL WHEEZY");
+            }
+          }
+        }
+    ).open(player);
+
+  }
+
+  public void unsubscribeFromService(Player player, int id) {
+    Service service;
+    try {
+      service = databaseManager.getServiceDao().queryForId(id);
+    } catch (SQLException e) {
+      e.printStackTrace();
+      player.sendMessage(ChatColor.RED + "SQL EXCEPTION TELL WHEEZY");
+      return;
+    }
+    if (service == null) {
+      player.sendMessage(ChatColor.RED + "Service id not found");
+      return;
+    }
+
+    Subscription subscription = service.getSubscription(player);
+    if (subscription == null) {
+      player.sendMessage(ChatColor.RED + "You are not subscribed to this service");
+      return;
+    }
+
+    new ConfirmationGui.Builder().title("Unsubscribe from " + service.getName() + "?")
+        .onChoiceMade(c -> {
+          if (!c) return;
+          try {
+            databaseManager.getSubscriptionDao().delete(subscription);
+            player.sendMessage(ChatColor.GREEN + "You have unsubscribed from " + service.getName() +
+                " (from " + service.getCompany().getName() + ")");
+          } catch (SQLException e) {
+            e.printStackTrace();
+            player.sendMessage(ChatColor.RED + "SQL EXCEPTION TELL WHEEZY");
+          }
+        }).open(player);
+
+  }
+
+  public void createService(Player player, double duration, double cost, int maxSubs, String name, Company company, AccountLink account) {
+    //Check for the same name
+    for (Service service : company.getServices()) {
+      if (service.getName().equals(name)) {
+        player.sendMessage(ChatColor.RED + "A service with the same name already exists for this company");
+        return;
+      }
+    }
+
+    if (duration <= 0.5) {
+      player.sendMessage(ChatColor.RED + "Service duration must be greater than 0.5 (12 hours)");
+      return;
+    }
+
+    if (name.length() > 40) {
+      player.sendMessage(ChatColor.RED + "Service name cannot be longer than 40 characters");
+      return;
+    }
+    Service newService = new Service(name, company, account, duration, cost, maxSubs);
+    try {
+      databaseManager.getServiceDao().create(newService);
+      player.sendMessage(ChatColor.GREEN + "Service created!");
+    } catch (SQLException e) {
+      e.printStackTrace();
+      player.sendMessage(ChatColor.RED + "SQL EXCEPTION TELL WHEEZY");
+    }
+  }
+
+  public void subscribeToService(Player player, int serviceId, Boolean autoPay) {
+    Service service;
+    try {
+      service = databaseManager.getServiceDao().queryForId(serviceId);
+    } catch (SQLException e) {
+      e.printStackTrace();
+      player.sendMessage(ChatColor.RED + "SQL EXCEPTION TELL WHEEZY");
+      return;
+    }
+    if (service == null) {
+      player.sendMessage(ChatColor.RED + "That service id does not exist");
```

Comment:
Use `sendMessage(Player, String)` and don't use `ChatColor.RED` (Also follow the messages i've used for errors)